### ISSUE TITLE
CAPZ with ASO doesn't work for AzureUSGovernment

### ISSUE THREAD

**Issue by ionutleca** [kubernetes-sigs/cluster-api-provider-azure#4113]
/kind bug

**What steps did you take and what happened:**
The [AzureManagedControlPlane](https://capz.sigs.k8s.io/reference/v1beta1-api#infrastructure.cluster.x-k8s.io/v1beta1.AzureManagedControlPlane) supports the following parameter:
```
spec:
  azureEnvironment: AzureUSGovernmentCloud
 ```

The [createSecretFromClusterIdentity](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/asosecret_controller.go#L259) function doesn't set [azureResourceManagerEndpoint](https://github.com/Azure/azure-service-operator/blob/main/v2/charts/azure-service-operator/values.yaml#L53C1-L53C29) and any other cloud specific variables.

The ASO controller fails with:
`The subscription '***' could not be found.: PUT https://management.azure.com/subscriptions/***/resourceGroups/***`

**What did you expect to happen:**
CAPZ to also add to the `*-aso-secret` the values specific to what `spec.azureEnvironment` on the `AzureManagedControlPlane` resource points to.

**Anything else you would like to add:**
-

**Environment:**

- cluster-api-provider-azure version: v1.11.1
- Kubernetes version: (use `kubectl version`): v1.27.3
- OS (e.g. from `/etc/os-release`): AKSUbuntu-2004gen2fipscontainerd-202309.06.0

**Comment by nojnhuh** [comment]
It appears ASO can only configure the ARM endpoint globally for all resources it manages. I've opened an issue here to make that capability accessible at the per-resource level to match CAPZ's capabilities from before: https://github.com/Azure/azure-service-operator/issues/3447.

If your management cluster is only managing workload clusters for one cloud, you could possibly modify the ASO deployment manually to configure it as you need as a stopgap.

**Comment by nojnhuh** [comment]
Even if changing it in aso-controller-settings doesn't work, I suppose it would be possible to edit the ASO deployment and hardcode the value there or referring to a different secret instead of deriving it from the global ASO secret.

**Comment by nojnhuh** [comment]
@ionutleca Did that workaround work for you? I'll reopen this to keep tracking making this more automatic.

**Comment by ionutleca** [comment]
Yes, I managed to make it work with the right values set in `aso-controller-settings`. Thank you!

**Comment by nojnhuh** [comment]
Just opened this PR to at least allow setting these fields with environment variables when CAPZ is installed. Hopefully that's at least a step in the right direction: #4390

**Comment by nojnhuh** [comment]
There's still more to iron out here re: CAPZ configuring the environment per-workload cluster and ASO only configuring it globally for all resources it manages, but I don't think I'll be able to follow up with that during this milestone.

/unassign

**Comment by dtzar** [comment]
Blocked until https://github.com/Azure/azure-service-operator/issues/3447
